src/Views/CountDownView.vala

```diff
@@ -98,11 +101,14 @@ public class CountDownView : Gtk.Box {
                 return false;
             }
 
-            return is_paused? false : true;
+            return true;
         });
     }
 
     private void cancel_countdown () {
+        // Immediately stop the countdown Timeout
+        Source.remove (countdown);
```

Comment:
I guess we should remove the countdown in some other ways because this code introduces a couple of bugs:

1. Click **Pause** and **Resume** quickly, the timer speeds up
2. Click **cancel** and then immediately *record* again, the timer is still speeded up.

With my solution that didn't happen.

Reply:
The solution is to check if the timer is active, and only in that case Source.remove it:
```
if (!is_paused) {
    Source.remove (countdown);
}
```

I don't see any GLib-CRITICAL with this solution.